core/src/main/java/io/aiven/klaw/service/SchemaOverviewService.java

```diff
@@ -152,9 +168,12 @@ private void updateAvroSchema(
 
             schemaDetails.add(schemaMap);
             schemaOverview.setSchemaExists(true);
-            // Set Promotion Details
-            processSchemaPromotionDetails(schemaOverview, tenantId, schemaEnv, kafkaEnvIds);
-            log.info("Getting schema details for: " + topicNameSearch);
+            if (requestorOwnsTopic(
+                topicNameSearch, schemaEnv.getAssociatedEnv().getId(), userName, tenantId)) {
+              // Set Promotion Details
+              processSchemaPromotionDetails(schemaOverview, tenantId, schemaEnv, kafkaEnvIds);
```

Comment:
so there would not be any FE changes i guess

Reply:
Correct no front end changes